core/src/main/java/io/kestra/core/tasks/storages/Deduplicate.java

```diff
@@ -0,0 +1,183 @@
+package io.kestra.core.tasks.storages;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import io.kestra.core.models.annotations.Example;
+import io.kestra.core.models.annotations.Plugin;
+import io.kestra.core.models.annotations.PluginProperty;
+import io.kestra.core.models.tasks.RunnableTask;
+import io.kestra.core.models.tasks.Task;
+import io.kestra.core.runners.RunContext;
+import io.kestra.core.serializers.JacksonMapper;
+import io.micronaut.core.util.functional.ThrowingFunction;
+import io.swagger.v3.oas.annotations.media.Schema;
+import jakarta.validation.constraints.NotNull;
+import lombok.Builder;
+import lombok.EqualsAndHashCode;
+import lombok.Getter;
+import lombok.ToString;
+import lombok.experimental.SuperBuilder;
+
+import java.io.*;
+import java.net.URI;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.HashMap;
+import java.util.Map;
+
+@Schema(
+    title = "Deduplicate a file by retaining only the latest row for each extracted key.",
+    description = """
+        The `Deduplicate` task involves reading the input file twice, rather than loading the entire file into memory.
+        The first iteration is used to build a deduplication map in memory containing the last lines observed for each key.
+        The second iteration is used to rewrite the file without the duplicates.
+        """
+)
+@Plugin(
+    examples = {
+        @Example(
+            full = true,
+            code = {
+                """
+                    tasks:
+                       - id: deduplicate
+                         type: io.kestra.core.tasks.storages.Deduplicate
+                         expr: ".key"
+                    """
+            }
+        )
+    }
+)
+@SuperBuilder
+@ToString
+@EqualsAndHashCode
+@Getter
+public class Deduplicate extends Task implements RunnableTask<Deduplicate.Output> {
+
+    @Schema(
+        title = "The file to be compacted.",
+        description = "Must be a `kestra://` storage URI."
+    )
+    @PluginProperty(dynamic = true)
+    @NotNull
+    private String from;
+
+    @Schema(
+        title = "The 'pebble' expression to be used for extracting a row key.",
+        description = "The 'pebble' expression can be used for constructing a composite key."
+    )
+    @PluginProperty(dynamic = false)
```

Comment:
```suggestion
    @PluginProperty
```

Should also be mandatory I think